### DISCUSSION TITLE
`watch` doesn't trigger IDE changes on VSCode

### DISCUSSION THREAD

**Discussion by nandorojo** [dotansimha/graphql-code-generator#8544] [General] [answered]
Hey there! I'm using the new `client-preset` with `--watch`.

One issue I'm having: VSCode doesn't update types for Intellisense when the file is updated in the background. In order to see my changes propagated as types when I write `graphql(...)`, I have to first open the codegen'd file. Is there a workaround for this?

This is what I do when I'm not in watch mode. However, it kinda sucks. I open the file whenever I codegen.

```json
{
   "scripts": "yarn graphql-codegen && code ./gql/gql"
}
```

**Comment by charlypoly** [comment]
Hi @nandorojo,

Yes, this is a TypeScript limitation that we plan to solve by providing IDE plugins as discussed here: https://github.com/dotansimha/graphql-code-generator/discussions/8345

**Reply by nandorojo** [reply]
Thanks @charlypoly! Really appreciate the answer. I know this isn't a great question for open source, but do you have a sense of when this might get solved? Could I possibly sponsor some sort of bug bounty to give that item priority?

Also, while we're on the topic of IDEs...do you have a recommendation for a good syntax highlighting plugin for VSCode other than Apollo's? I'm using Apollo successfully, however it only supports a single graphql endpoint. I have multiple endpoints and haven't been able to get a syntax highlighter working in `.ts,.tsx` files using the client preset's `graphql()` function. Thanks!

**Comment by nandorojo** [comment, marked as answer]
I believe I found a solution. I shared it at https://github.com/dotansimha/graphql-code-generator/discussions/8345#discussioncomment-4028928
